BJS2-77538 | Add premium bought event analytics model and Kafka listener

Raidanmark opened the pull request:
Add PremiumBoughtEvent and related Kafka topic configuration

Raidanmark committed:
feat: Add PremiumBoughtEvent and related Kafka topic configuration

danbg12 reviewed (commented):
Четко! Только закоменченныи код убери :D

AlexeyAntipin commented on src/main/java/school/faang/user_service/kafka/events/PremiumBoughtEvent.java:
```diff
@@ -0,0 +1,21 @@
+import java.time.LocalDateTime;
+
+@Data
+//@Builder
```
Либо удаляй, либо оставляй

AlexeyAntipin commented on src/main/java/school/faang/user_service/kafka/producer/KafkaTopics.java:
```diff
@@ -24,4 +24,7 @@ public class KafkaTopics {
     private String recommendationRequestTopic;
+
+    @Value("${spring.kafka.topics.premium-bought-topic.name}")
+    private String premiumBoughtTopic;
```
Уже много топиков накопилось, давай класс отрефакторим и сделаем его @ConfigurationProperties

Raidanmark replied:
Что то я за это сел и запутался

AlexeyAntipin replied:
Почитай статью
https://habr.com/ru/articles/740802/

AlexeyAntipin commented on src/main/java/school/faang/user_service/service/premium/PremiumService.java:
```diff
@@ -25,6 +35,20 @@ public TransactionResultDto buyPremium(Long userId, Integer durationDays) {
                 .equals(TransactionStatus.SETTLED)) {
             premiumServiceUtils.assignPremiumToUser(userId, durationDays);
         }
+//        PremiumBoughtEvent premiumBoughtEvent = PremiumBoughtEvent.builder()
```
Надо убрать

AlexeyAntipin commented on src/main/resources/application.yaml:
```diff
@@ -93,6 +101,7 @@ spring:
         partitions: 4
         replication-factor: 3
 
+
```
Лишняя пустая строка

AlexeyAntipin commented on src/test/java/school/faang/user_service/service/premium/PremiumServiceTest.java:
```diff
@@ -1,66 +1,66 @@
-    }
-}
\ No newline at end of file
+//package school.faang.user_service.service.premium;
```
Страшно

Raidanmark committed:
refactor: Replace KafkaTopics with KafkaTopicsProperties in premium-related services and tests

AlexeyAntipin commented:
И нужно поправить конфликты

Raidanmark committed:
Refactor Kafka topic configuration and related usages.

Replaced plain string topic names with a structured `KafkaTopics.Topic` class to enhance type safety and better manage topic configurations. Updated all relevant services, tests, and YAML settings to align with the new structure.

Raidanmark committed:
Update property keys for recommendation request topic

Renamed Kafka topic property keys to align with the naming convention. This ensures consistency and avoids potential configuration mismatches. No functional changes were made.

Raidanmark committed:
Merge branch 'hydra-master-stream10' into BJS2-77538

# Conflicts:
#	src/main/java/school/faang/user_service/kafka/producer/DataSender.java
#	src/main/java/school/faang/user_service/kafka/producer/KafkaTopics.java
#	src/main/java/school/faang/user_service/kafka/producer/KafkaTopicsConfig.java
#	src/main/resources/application.yaml

Raidanmark committed:
Conflicts resolving

Refactor Kafka topic naming and improve type safety in sender.

Updated Kafka topic configurations and naming conventions for better consistency and clarity. Modified `send` methods to use `KafkaTopics.Topic` type, enhancing type safety and reducing potential errors.

Raidanmark committed:
test fixes

Raidanmark committed:
test fixes
